torch/csrc/utils/schema_info.cpp

```diff
@@ -67,7 +67,26 @@ bool SchemaInfo::is_mutable(size_t index) {
   return std::any_of(
       input_alias_map_[index].begin(),
       input_alias_map_[index].end(),
-      [this](size_t index) { return this->schema_.is_mutable(index); });
+      [this](size_t index) {
+        static const char* batch_norm_literal =
```

Comment:
A few suggestions here:

1. Can you do the schema parsing statically so that it doesn't have to happen each time?
2. Can you move the special case behavior outside of the `std::any_of`?  The special case behavior doesn't really have any dependence on index, it would probably be easier to understand just as `if (this_schema == batch_norm_literal && (has_running_mean || has_running_var) { return training_is_true; }`

Reply:
1. Yeah I'll clean that up and also make the special case look more readable.
2. Technically there actually is a dependence on index. For instance, with batch_norm if the input tensor value is aliasing the running_mean and training = true, then the input tensor will now be mutable which is found through std::any_of